app/util/FeatureFlagSupport.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package util
+
+import config.AppConfig
+import play.api.mvc.Result
+import play.api.mvc.Results.NotFound
+
+trait FeatureFlagSupport {
+
+  def ifPartialsEnabled(block: Result)(implicit appConfig: AppConfig): Result =
+    if (appConfig.disablePartials) {
+      NotFound("")
```

Comment:
should we [410 Gone](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/410) it and put a message in the response body like `Loading contact-frontend forms via partials is deprecated, see #event-deprecate-partial-contact-forms`

Reply:
I was trying to think of a good message and then decided not to have one because my thought was if this feature switch was enabled in production and for some reason, a team never removed partials then the public user would see this message.

Reply:
I didn't think of that, good catch! It looks like assets-frontend ajax call will [show a fixed error - but only if the responseBody is empty otherwise it seems to show nothing](https://github.com/hmrc/assets-frontend/blob/f0b2b9830405199345e0f7b5abd710348b14cc8c/assets/javascripts/modules/reportAProblem.js#L52C35-L52C35) and I couldn't find an example of a contact form loaded by partial anymore but it seems like it doesn't use the error response body by default https://github.com/hmrc/play-partials/blob/main/src/main/scala/uk/gov/hmrc/play/partials/HtmlPartial.scala#L38C42-L38C57 so it would just be some member of the public calling the partial url by itself which could see the error message and I don't think that's a problem - but for the first case of assets-frontend it makes sense to leave it blank, but I reckon it should be 410 rather than 404 as a status code